src/pattern-library/util/jsx-to-string.js

```diff
@@ -8,11 +8,11 @@ function escapeQuotes(str) {
 }
 
 function componentName(type) {
-  if (typeof type === 'string') {
-    return type;
-  } else {
-    return type.displayName || type.name;
-  }
+  const name = typeof type === 'string' ? type : type.displayName ?? type.name;
+  // Handle (display)name conflicts if there are two components with the same
+  // name. e.g. if there are two components named `Foo`, the second of those
+  // encountered will have a name of `Foo$1`. Strip the `$1` in this case.
+  return name.replace(/\$[0-9]+$/, '');
```

Comment:
This is real quick-and-dirty, but, it works, and is only needed until we don't have duplicate components with the same names in the package.